### PR TITLE
[ISSUE #9391] Chooser optimization

### PR DESCRIPTION
**Pull request by RapperCL**
[issue](https://github.com/alibaba/nacos/issues/9391)

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/naming/utils/Chooser.java [hunk 1]
```diff
@@ -143,23 +143,20 @@ public void refresh() {
...
             for (Pair<T> item : itemsWithWeight) {
                 double singleWeight = item.weight();
                 //ignore item which weight is zero.see test_randomWithWeight_weight0 in ChooserTest
                 if (singleWeight <= 0) {
                     continue;
                 }
-                exactWeights[index++] = singleWeight / originWeightSum;
-            }
-            
-            weights = new double[items.size()];
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/utils/Chooser.java:158]: I read this codes, I think this array size might be `index` no items.size()

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
I'm contacting the author for this part codes. After making sure the codes meaning, I will response this PR and determine whether change or how to change this codes. Before making sure, The PR will be pending, Sorry.

**Review by PRANRAVEE — APPROVED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
Can you add some Unit test to ensure your changes has no effect for this method?

**Comment by RapperCL** [comment]
> Can you add some Unit test to ensure your changes has no effect for this method?

OK, I'll add it later

**Comment by KomachiSion** [comment]
> I'm contacting the author for this part codes. After making sure the codes meaning, I will response this PR and determine whether change or how to change this codes. Before making sure, The PR will be pending, Sorry.

I talk about this problem with author, he think the length of weight also should be `index`. So I think after your unit test finished, the PR can be merged.

**Comment by RapperCL** [comment]
> > I'm contacting the author for this part codes. After making sure the codes meaning, I will response this PR and determine whether change or how to change this codes. Before making sure, The PR will be pending, Sorry.
> 
> I talk about this problem with author, he think the length of weight also should be `index`. So I think after your unit test finished, the PR can be merged.

Well, thank you. Interestingly, when I was writing the test case, I found that the chooser would have an exception in a special scenario. In order to prevent confusion, I raised an issue again [issue 9478](https://github.com/alibaba/nacos/issues/9478) and the corresponding pr submitted the solution and test case.

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**